### PR TITLE
[5.5] Build PackageDescription and PackagePlugin universal on macOS platform

### PR DESCRIPTION
**Pull request by abertelrud**
This is the 5.5 cherry-pick of a sequence of changes to build PackageDescription and PackagePlugin universal on macOS platform.  SwiftPM switches to the XCBuild build system when building universal, and this PR includes the necessary follow-up changes to make that work.

This includes 5.5 cherry-picks of the changes in:

https://github.com/apple/swift-package-manager/pull/3431
https://github.com/apple/swift-package-manager/pull/3526
https://github.com/apple/swift-package-manager/pull/3528

each as a separate commit containing the original commit hash from which it was cherry-picked.

This PR builds on top of the PR https://github.com/apple/swift-package-manager/pull/3492, which is kept as a separate PR since it is for a different issue.

The three are put up as one PR since they all are steps toward fixing the same issue.

### CODE REVIEW

**Comment by abertelrud** [comment]
@swift-ci please test

**Review by tomerd — APPROVED** [review]

**Comment by abertelrud** [comment]
@swift-ci please smoke test

**Comment by abertelrud** [comment]
@swift-ci please test

**Review by neonichu — APPROVED** [review]

**Comment by abertelrud** [comment]
@swift-ci please test

**Comment by abertelrud** [comment]
@swift-ci please test linux

**Comment by buttaface** [comment]
Anything holding this up? My cross-compilation pull #3577 relies on this pull to cross-compile these libraries.

**Comment by abertelrud** [comment]
> Anything holding this up? My cross-compilation pull #3577 relies on this pull to cross-compile these libraries.

No, it looks like this has all the necessary approvals.  Merging.

**Pull request merged**
